org-ql.el

```diff
@@ -938,10 +950,20 @@ Arguments STRING, POS, FILL, and LEVEL are according to
               :action (lambda ()
                         (throw 'found t)))))))))
 
+(org-ql--defpred parent (querypredicatefn)
+  "Return non-nil if the current entry's parent matches the compiled QUERYPREDICATEFN."
+  (save-excursion
+    (save-restriction
+      (widen)
+      (when (org-up-heading-safe)
+	(--when-let (org-ql--value-at (point) querypredicatefn)
+	  (not (eq 'org-ql-nil it)))))))
```

Comment:
`org-ql--value-at` does not return `org-ql-nil`.  That symbol is used by it internally.

Reply:
Hmm. Somehow this doesn't work without checking for `org-ql-nil`. It looks like `org-ql--value-at` returns `new-value` which is set to `org-ql-nil` if the predicate returns `nil`.

Reply:
Okay, I fixed an issue where `org-ql--value-at` will return `org-ql-nil` when there's a cache miss and the `fn` returns `nil`.